### ISSUE TITLE
KOSMOS-2, finding probability distribution of the text sequence

### ISSUE THREAD

**Issue by snpushpi** [huggingface/transformers#28619]
### System Info

Google Colab with gpu environment enabled and related libraries installed  

### Who can help?

@ydshieh 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below

### Reproduction

Hi, 
I am trying to extract the probability distribution of the tokens in a sequence from the model. Here is what 
`
from PIL import Image
import requests
from transformers import AutoProcessor, Kosmos2ForConditionalGeneration
import torch

model = Kosmos2ForConditionalGeneration.from_pretrained("microsoft/kosmos-2-patch14-224")
processor = AutoProcessor.from_pretrained("microsoft/kosmos-2-patch14-224")

prompt1 = "An old man sitting on a bench in a public park alone, reading a book."
url1 = "http://images.cocodataset.org/val2017/000000264535.jpg"
image1 = Image.open(requests.get(url1, stream=True).raw)

inputs1 = processor(text=prompt1, images=image1, return_tensors="pt").to(device)

model_output = model(
    pixel_values=inputs1["pixel_values"],
    input_ids=inputs1["input_ids"],
    image_embeds=None,
    image_embeds_position_mask=inputs1["image_embeds_position_mask"],
    use_cache=True,
)
input_ids = processor.tokenizer(prompt, return_tensors = 'pt').input_ids.to(device)
input_ids.shape
# torch.Size([1, 14])

model_output.logits.shape 
# torch.Size([1, 79, 65037])
token_logits = torch.cat([model_output.logits[:,:2,:],model_output.logits[:,67:,:]],dim=1) #Is this correct?
`
So I am trying to extract the logits of the text tokens from the model output. The goal is to eventually calculate the probability of a certain word given the previous words and the image. But I am not sure of the last line where I extracted the text token logits from the model output. I did what I did since that's what the inputs1["image_embeds_position_mask"] mask looked like, but I am not sure if that is the right thing to do. Can someone confirm that?

### Expected behavior

explained above.

**Comment by snpushpi** [comment]
@ydshieh can you look at it? Thank you so much!

**Comment by ydshieh** [comment]
Hi, 

I think you are only interested in looking the logits for the parts of `prompt1 = "An old man sitting on a bench in a public park alone, reading a book."`.

This part is after the token (id) of `end of image` (eoi, `64004`). So you have to find where the token id `64004` is in `inputs1["input_ids"]`, and only take the parts after it (along the sequence dimension).

- using `image_embeds_position_mask` works, but you don't need to having the leading 2 tokens
  - using `image_embeds_position_mask` includes `64004`, and the logits from that place is the logit for the next token (`An`) given the token of `eoi (64004)`.

**Comment by snpushpi** [comment]
Thank you, that makes sense!
